dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/EventHookControllerTest.java

```diff
@@ -0,0 +1,156 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller;
+
+import static org.hisp.dhis.web.WebClient.Body;
+import static org.hisp.dhis.web.WebClientUtils.assertStatus;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.hisp.dhis.eventhook.targets.WebhookTarget;
+import org.hisp.dhis.eventhook.targets.auth.ApiTokenAuth;
+import org.hisp.dhis.eventhook.targets.auth.HttpBasicAuth;
+import org.hisp.dhis.jsontree.JsonList;
+import org.hisp.dhis.jsontree.JsonObject;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.webapi.DhisControllerIntegrationTest;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the {@link EventHookController} using (mocked) REST requests.
+ *
+ * @author Morten Olav Hansen
+ */
+class EventHookControllerTest extends DhisControllerIntegrationTest
+{
+    @Test
+    void testGetEventHooks()
+    {
+        JsonObject eventHooks = GET( "/eventHooks" ).content( HttpStatus.OK );
+        assertTrue( eventHooks.isObject() );
+        assertTrue( eventHooks.has( "eventHooks" ) );
+        assertTrue( eventHooks.get( "eventHooks" ).isArray() );
+        assertTrue( eventHooks.get( "eventHooks" ).asList( JsonObject.class ).isEmpty() );
+    }
+
+    @Test
+    void testCreateEventHookWebhookApiToken()
+    {
+        String id = assertStatus( HttpStatus.CREATED,
+            POST( "/eventHooks", Body( "event-hook/webhook-api-token.json" ) ) );
+        assertEquals( "bRNvL6NMQXb", id );
+
+        JsonObject eventHook = GET( "/eventHooks/{id}", id ).content( HttpStatus.OK );
+        assertTrue( eventHook.has( "id", "name", "source", "targets" ) );
+        assertEquals( "bRNvL6NMQXb", eventHook.getString( "id" ).string() );
+        assertEquals( "WebhookApiToken", eventHook.getString( "name" ).string() );
+        assertEquals( "metadata", eventHook.get( "source" ).asObject().getString( "path" ).string() );
+        assertEquals( "id,name", eventHook.get( "source" ).asObject().getString( "fields" ).string() );
+
+        JsonList<JsonObject> targets = eventHook.get( "targets" ).asList( JsonObject.class );
+        assertFalse( targets.isEmpty() );
+
+        JsonObject target = targets.get( 0 ).asObject();
```

Comment:
`getObject(0)` instead of `get( 0 ).asObject()` is less "casty". There are a couple more of similar cases where `getObject` could be used instead.